src/libtls/tests/suites/test_socket.c

```diff
@@ -254,6 +335,78 @@ START_SETUP(setup_creds)
 }
 END_SETUP
 
+START_SETUP(setup_ed25519_creds)
```

Comment:
I think this could be simplified with a helper function that takes the ECDSA/EdDSA key and certificate chunks (and maybe the key type, although, KEY_ANY should also work). So all setup_creds functions would then be simple calls of that shared code.

Also, what happens if no plugin provides EdDSA (or ECDSA actually)?